quesma/queryparser/pancake_transformer.go

```diff
@@ -137,6 +140,20 @@ func (a *pancakeTransformer) createLayer(previousAggrNames []string, childAggreg
 				result[0].layer.nextBucketAggregation = bucket
 				result[0].nextBucketAggregation = childAgg
 			} else {
+				// if both leaf optimizations are filter and second one doesn't have children we can treat second as metric
```

Comment:
I don't like, but without that we generate two SQLs for sample dashboards.

Reply:
This can be a part of the "pancake" optimization pipeline. Or it can be moved to the separate "optimizeSth()" function.